docs/src/assets/scss/utilities.scss

```diff
@@ -19,7 +19,7 @@
 }
 
 [hidden] {
-    display: none !important;
+    display: none;
```

Comment:
I believe this change causes the fallback Versions link to appear on mobile even when JavaScript is enabled because [`.docs-site-nav a:not(.c-btn)`](https://github.com/eslint/eslint/blob/6cca82432ad7b74b5ec96126ba467707a8884320/docs/src/assets/scss/components/docs-navigation.scss#L34-L43) has a higher specificity.

<details>
<summary> mobile nav </summary>

![image](https://user-images.githubusercontent.com/44349756/171065122-f09cdc82-6568-4de2-b5f6-043642699722.png)

</details>

Reply:
Nice catch. This is actually a bug on the prototype site too.